src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -60,19 +60,32 @@ def completer(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-arg
             return [r.name for r in get_resources_in_subscription(resource_type=resource_type)]
     return completer
 
-def get_enum_type_completion_list(enum_type=None):
-    ENUM_MAPPING_KEY = '_value2member_map_'
-    def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
-        return list(enum_type.__dict__[ENUM_MAPPING_KEY].keys())
-    return completer
-
 def get_generic_completion_list(generic_list):
     def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
         return generic_list
     return completer
 
-def get_enum_choices(enum_type):
-    return [x.value for x in enum_type]
+class CaseInsenstiveList(list): # pylint: disable=too-few-public-methods
```

Comment:
Here's a real CLI example:

Case-preserving
```
 --load-distribution           : Affinity rule settings.  Allowed values: Default, SourceIP,
                                    SourceIPProtocol.
```
Forced-lower
```
 --load-distribution           : Affinity rule settings.  Allowed values: default, sourceip,
                                    sourceipprotocol
```
To me, the first is more readable than the second. Either format will be accepted, or a crazy mixed case (SoUrCeIpPrOtOcOl). Perhaps forcing to to camel case (i.e. making sure the first letter is always lowercase) would preserve the readability and keep it consistent. However, we have plenty of places where our choice lists contain uppercase.

@derekbekoe @yugangw-msft @burtbiel @haocs @mayurid @jasonrshaver  Thoughts?